Paco/src/com/pacoapp/paco/model/ExperimentProviderUtil.java

```diff
@@ -780,6 +811,46 @@ public void insertEvent(EventInterface eventI) {
     }
   }
 
+  public List<Event> findEventsByCriteriaQuery(String[] projection, String criteriaColumns, String criteriaValues[], String sortOrder, String limit, String groupBy, String having) {	    
+	    Cursor cursor = null;
+	    List<Event> events = null;
+	    DatabaseHelper dbHelper = new DatabaseHelper(context);
+	    if(sortOrder != null){
+	    	sortOrder = sortOrder.concat(LIMIT).concat(limit);
+	    }else{
+	    	sortOrder = LIMIT.concat(limit);
+	    }
+	    try {
+	    	List<String> allColumns = Arrays.asList(projection);
+	    	String tableIndicator = ExperimentUtil.identifyTablesInvolved(eventsOutputColumns, allColumns);
+	    	if(tableIndicator.equals(ExperimentProvider.EVENTS_OUTPUTS_TABLE_NAME)){
+	    		cursor = dbHelper.query(ExperimentProvider.EVENTS_OUTPUTS_DATATYPE, projection, criteriaColumns, criteriaValues, sortOrder, groupBy, having);
+	    	}else if(tableIndicator.equals(ExperimentProvider.EVENTS_TABLE_NAME)){
+	    		cursor = dbHelper.query(ExperimentProvider.EVENTS_DATATYPE,projection, criteriaColumns, criteriaValues, sortOrder, groupBy, having);
+	    	}else if(tableIndicator.equals(ExperimentProvider.OUTPUTS_TABLE_NAME)){
+	    		cursor = dbHelper.query(ExperimentProvider.OUTPUTS_DATATYPE, projection, criteriaColumns, criteriaValues, sortOrder, groupBy, having);
+	    	}
+	    		    	
+	    	if (cursor != null) {
+	    		events = new ArrayList<Event>();
```

Comment:
For new Collections, please use Lists.newArrayList() 
(fyi, its in the google collections library and is available on the classpath already)